### PR TITLE
Fix running on Linux

### PR DESCRIPTION
**Pull request by ankon**
Bunch of changes I needed to get some sane output on Linux (Fedora 40/x86_64, specifically)

### CODE REVIEW

**Comment by ankon** [comment]
One thing that I still needed to do for larger sites (>1M!): bump the time for waiting for tcpdump to flush things from 1s to 10s. I guess for this tools things don't have to be super-duper-fancy, so simply doing that by default is probably a good idea.

WDYT?

File: README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ reason. Contributions encouraged!
 # You may need to install libpcap development packages for packetfu to work,
 # depending on your operating system.
 
-sudo gem install packetfu net-ping
+sudo bundle install
```
Comment (sirupsen) [inline README.md:16]: 👍🏻

File: initcwnd.rb [hunk 2]
```diff
@@ -56,12 +56,15 @@
 # The reason I didn't is because curl supports SSLKEYLOGFILE which is very
 # convenient for Wireshark.
 # TODO: We do not send --compressed
-site_html = `SSLKEYLOGFILE=key.log curl --tlsv1.2 -k -H "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36" --fail --http1.1 #{uri.to_s}`
+# TODO: Support TLS 1.3
+site_html = `SSLKEYLOGFILE=key.log curl --tlsv1.2 --tls-max 1.2 -k -H "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36" --fail --http1.1 #{uri.to_s}`
```
Comment (sirupsen) [inline initcwnd.rb:60]: Did you need `--tls-max` here because it doesn't work with the key log file for 1.3?

File: initcwnd.rb [hunk 3]
```diff
@@ -47,7 +47,7 @@
 File.delete("initcwnd.pcap") if File.exist?("initcwnd.pcap")
 
 pid = fork do
-  exec("tcpdump", "-i", interface, "-w", "initcwnd.pcap", "host", uri.host, "and", "port", "443")
+  exec("tcpdump", "-i", interface, "-w", "initcwnd.pcap", "--packet-buffered", "host", uri.host, "and", "port", "443")
```
Comment (sirupsen) [inline initcwnd.rb:50]: Why did you need this? Does it work on MacOS?

**Review by sirupsen — COMMENTED** [review]
Hey, sorry for the delay in review!

File: initcwnd.rb (hunk 3 above)
Comment (ankon) [reply initcwnd.rb:50]: From the commit message:
> Use `--packet-buffered` so that tcpdump will write packets into the dump as it processes
  them, rather than waiting for a buffer to fill up.

This is a standard option of tcpdump, so while I didn't try I would assume it works on macOS as well.

File: initcwnd.rb (hunk 2 above)
Comment (ankon) [reply initcwnd.rb:60]: Indeed: if curl was able to do TLS 1.3 the logic in the script wouldn't work and produce no useful output. 

I think having this strict here to show it works under certain known conditions and not under others will make it easier for other people to pick up the needed work :)
